Objects/typeobject.c

```diff
@@ -4747,6 +4877,82 @@ is_dunder_name(PyObject *name)
     return 0;
 }
 
+static void
+update_cache(struct type_cache_entry *entry, PyObject *name, unsigned int version_tag, PyObject *value)
+{
+    entry->version = version_tag;
+    entry->value = value;  /* borrowed */
```

Comment:
We will want to use relaxed atomic writes for version, value, and name because reads can happen concurrently.